schema.c

```diff
@@ -445,9 +479,60 @@ sch_load_namespace_mappings (sch_instance *instance, const char *filename)
     }
 }
 
+static void
+sch_load_model_list (sch_instance *instance, const char *path, const char *model_list_filename)
+{
+    FILE *fp = NULL;
+    char *name;
+    char buf[256];
+
+    if (!instance)
+        return;
+
+    name = g_strdup_printf ("%s/%s", path, model_list_filename);
+    fp = fopen (name, "r");
+    if (fp)
+    {
+        if (!instance->model_hash_table)
+            instance->model_hash_table = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_free);
+
+        while (fgets (buf, sizeof (buf), fp) != NULL)
+        {
+            /* Skip comment lines */
+            if (buf[0] == '#')
+                continue;
+
+            /* Remove any trailing LF */
+            buf[strcspn(buf, "\n")] = '\0';
+            if (strlen (buf) > 0)
+            {
+                void *old_key;
+                void *old_value;
+
+                /* Look up this node name to check for duplicates. */
```

Comment:
is this block correct? you seem to be taking the same action in both cases. no point in doing the lookup then?

Reply:
Fixed